### PR TITLE
Save raw output data from execution events

### PR DESCRIPTION
**Pull request by katrogan**
Signed-off-by: Katrina Rogan <katroganGH@gmail.com>

# TL;DR
This change handles receiving raw output data for successful execution events and saving that inline. As part of this change, we also update the observations for output data size and can use this to inform whether to offload output data in the future if the size ends up being particularly large in practice and effects db performance.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1327

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/manager/impl/execution_manager.go [hunk 1]
```diff
@@ -1188,6 +1188,9 @@ func (m *ExecutionManager) CreateWorkflowEvent(ctx context.Context, request admi
 		m.systemMetrics.ActiveExecutions.Dec()
 		m.systemMetrics.ExecutionsTerminated.Inc()
 		go m.emitOverallWorkflowExecutionTime(executionModel, request.Event.OccurredAt)
+		if request.Event.GetOutputData() != nil {
```
Comment (EngHabu) [inline pkg/manager/impl/execution_manager.go:1191]: Should we check/validate the size and fail the request if it's bigger than "maxDataSize" used later when retrieving executions?

Comment (katrogan) [reply pkg/manager/impl/execution_manager.go:1191]: oh you mean and offload the outputs at this time @EngHabu

Comment (katrogan) [reply pkg/manager/impl/execution_manager.go:1191]: ping @EngHabu

Comment (EngHabu) [reply pkg/manager/impl/execution_manager.go:1191]: Sorry just saw this.. Either offload or fail the request...

File: pkg/manager/impl/execution_manager.go [hunk 2]
```diff
@@ -1286,16 +1289,24 @@ func (m *ExecutionManager) GetExecutionData(
 	}
 	if remoteDataScheme == common.Local || remoteDataScheme == common.None || (signedOutputsURLBlob.Bytes < maxDataSize && execution.Closure.GetOutputs() != nil) {
 		var fullOutputs core.LiteralMap
-		outputsURI := execution.Closure.GetOutputs().GetUri()
-		err := m.storageClient.ReadProtobuf(ctx, storage.DataReference(outputsURI), &fullOutputs)
-		if err != nil {
-			logger.Warningf(ctx, "Failed to read outputs from URI [%s] with err: %v", outputsURI, err)
+		if execution.Closure.GetOutputs() != nil {
```
[the remark this answers is not in the document]
Comment (EngHabu) [reply pkg/manager/impl/execution_manager.go:1292]: 🤐

**Review by EngHabu — APPROVED** [review]

**Comment by katrogan** [comment]
tested with a flyte deployment with propeller set to send event data both inline and by reference and verified that calling Get*ExecutionData return the full inputs and outputs for both
